core/src/main/java/com/linecorp/armeria/common/util/ThreadFactoryBuilder.java

```diff
@@ -0,0 +1,109 @@
+    private PentaFunction<String, Boolean, Integer, ThreadGroup,
+                          Function<? super Runnable, ? extends Runnable>,
+                          ? extends AbstractThreadFactory> factoryConstructor;
+
+    /**
+     * Creates a new factory builder that creates a specified type of {@link ThreadFactory}.
+     *
+     * @param threadNamePrefix the prefix of the names of the threads created by this factory.
+     * @param factoryConstructor the constructor reference of concrete {@link AbstractThreadFactory}.
+     */
+    public ThreadFactoryBuilder(String threadNamePrefix,
+                                PentaFunction<String, Boolean, Integer, ThreadGroup,
+                                              Function<? super Runnable, ? extends Runnable>,
+                                              ? extends AbstractThreadFactory> factoryConstructor) {
+        this.threadNamePrefix = requireNonNull(threadNamePrefix, "threadNamePrefix");
+        this.factoryConstructor = requireNonNull(factoryConstructor, "factoryConstructor");
+    }
+
+    /**
+     * Sets daemon for new threads.
+     */
+    public ThreadFactoryBuilder daemon(boolean daemon) {
+        this.daemon = daemon;
+        return this;
+    }
+
+    /**
+     * Sets priority for new threads.
+     */
+    public ThreadFactoryBuilder priority(int priority) {
+        if (priority < Thread.MIN_PRIORITY || priority > Thread.MAX_PRIORITY) {
+            throw new IllegalArgumentException(
+                    "priority: " + priority +
+                    " (expected: Thread.MIN_PRIORITY <= priority <= Thread.MAX_PRIORITY)");
+        }
+
+        this.priority = priority;
+        return this;
+    }
+
+    /**
+     * Sets thread group for new threads.
+     */
+    public ThreadFactoryBuilder threadGroup(ThreadGroup threadGroup) {
+        this.threadGroup = requireNonNull(threadGroup, "threadGroup");
+        return this;
+    }
+
+    /**
+     * Sets task function for new threads.
+     * Use this method to set additional work before or after the Runnable is run. For example:
+     * <pre>{@code
+     * ThreadFactories.builder("thread-prefix")
+     *                .taskFunction( task -> {
+     *                    return () -> {
+     *                        // Add something to do before task is run
+     *                           task.run();
+     *                        // Add something to do after task is run
+     *                    };
+     *                })
+     *                .build();
+     * }</pre>
+     */
+    public ThreadFactoryBuilder taskFunction(
+            Function<? super Runnable, ? extends Runnable> taskFunction) {
+        this.taskFunction = requireNonNull(taskFunction, "taskFunction");
+        return this;
+    }
+
+    /**
+     * Returns a new {@link ThreadFactory} instance.
+     */
+    public AbstractThreadFactory build() {
```

Comment:
Could return `ThreadFactory` and hide `AbstractThreadFactory` from the public API completely.

Reply:
Fixed it. Thank you.